cmd/npmExecuteLint_test.go

```diff
@@ -0,0 +1,363 @@
+					return err
+				}
+			}
+
+			err = n.utils.execRunner.RunExecutable("npm", npmRunArgs...)
+			if err != nil {
+				return fmt.Errorf("failed to run npm script %s: %w", script, err)
+			}
+		}
+	}
+	return nil
+}
+
+// InstallAllDependencies mocking implementation
+func (n *npmExecutorMock) InstallAllDependencies(packageJSONFiles []string) error {
+	for _, packageJSON := range packageJSONFiles {
+		dir := filepath.Dir(packageJSON)
+		if dir == "." {
+			dir = ""
+		} else {
+			dir = dir + string(os.PathSeparator)
+		}
+		if n.utils.HasFile(dir + "package-lock.json") {
+			err := n.utils.execRunner.RunExecutable("npm", "ci")
+			if err != nil {
+				return err
+			}
+		} else if n.utils.HasFile(dir + "yarn.lock") {
+			err := n.utils.execRunner.RunExecutable("yarn", "install", "--frozen-lockfile")
+			if err != nil {
+				return err
+			}
+		} else {
+			err := n.utils.execRunner.RunExecutable("npm", "install")
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// SetNpmRegistries mocking implementation
+func (n *npmExecutorMock) SetNpmRegistries() error {
+	const sapRegistry = "@sap:registry"
+	const npmRegistry = "registry"
+	configurableRegistries := []string{npmRegistry, sapRegistry}
+	for _, registry := range configurableRegistries {
+		var buffer bytes.Buffer
+		n.utils.execRunner.Stdout(&buffer)
+		err := n.utils.execRunner.RunExecutable("npm", "config", "get", registry)
+		preConfiguredRegistry := buffer.String()
+		if strings.HasPrefix(preConfiguredRegistry, "undefined") {
+			if registry == npmRegistry && n.options.defaultNpmRegistry != "" {
+				err = n.utils.execRunner.RunExecutable("npm", "config", "set", registry, n.options.defaultNpmRegistry)
+				if err != nil {
+					return err
+				}
+			}
+			if registry == sapRegistry && n.options.sapNpmRegistry != "" {
+				err = n.utils.execRunner.RunExecutable("npm", "config", "set", registry, n.options.sapNpmRegistry)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// FindPackageJSONFiles mocking implementation
+func (n *npmExecutorMock) FindPackageJSONFiles() []string {
+	var packageJSONFiles []string
+	unfilteredPackageJSONFiles, _ := n.utils.Glob("**/package.json")
+
+	for _, file := range unfilteredPackageJSONFiles {
+		if !(strings.Contains(file, "node_modules") || strings.HasPrefix(file, "gen"+string(os.PathSeparator)) || strings.Contains(file, string(os.PathSeparator)+"gen"+string(os.PathSeparator))) {
+			packageJSONFiles = append(packageJSONFiles, file)
+		}
+	}
+	return packageJSONFiles
+}
+
+func TestNpmExecuteLint(t *testing.T) {
+	t.Run("Call with ci-lint script and one package.json", func(t *testing.T) {
+		utils := newNpmMockUtilsBundle()
+		utils.AddFile("package.json", []byte("{\"scripts\": { \"ci-lint\": \"\" } }"))
+
+		config := npmExecuteLintOptions{}
+
+		npmExecutor := npmExecutorMock{utils: utils, options: npmExecuteOptions{
+			install:            false,
+			runScripts:         []string{"ci-lint"},
```

Comment:
I think also the definition of "ci-lint" should not be part of the test, this should be done by the step